packages/core/src/open.ts

```diff
@@ -43,6 +62,7 @@ async function open_v2<Store extends Readable>(
 	let loc = "store" in location ? location : new Location(location);
 	let attrs = {};
 	if (options.attrs ?? true) attrs = await load_attrs(loc);
+	VERSION_COUNTER.increment(loc.store, "v2");
```

Comment:
For v2, maybe we only increment if a node has successfully been found? this within `open_array_v2` and `open_group_v2` but only if they are successful?

```diff
async function open_array_v2<Store extends Readable>(
	location: Location<Store>,
	attrs: Attributes,
) {
	let { path } = location.resolve(".zarray");
	let meta = await location.store.get(path);
	if (!meta) {
		throw new NodeNotFoundError(path);
	}
++	VERSION_COUNTER.increment(loc.store, "v2");
	return new Array(
		location.store,
		location.path,
		v2_to_v3_array_metadata(json_decode_object(meta), attrs),
	);
}
```

Or we add some wrappers functions:

```ts
async function open_v2_with_counter<Store extends Readable>(...args: Parameters<typeof open_v2<Store>>) {
	let node = await open_v2(...args);
	VERSION_COUNTER.increment(loc.store, "v2");
	return node;
}
```